conjure-undertow-processor/src/main/java/com/palantir/conjure/java/undertow/processor/data/ParamTypesResolver.java

```diff
@@ -349,6 +354,32 @@ private CodeBlock getCollectionParamDecoder(
                 });
     }
 
+    private Optional<CodeBlock> getParamDecoderConstructor(DeclaredType declaredType) {
+        TypeElement typeElement = (TypeElement) declaredType.asElement();
+        // new T(String)
+        return typeElement.getEnclosedElements().stream()
+                .filter(element -> element.getKind() == ElementKind.CONSTRUCTOR)
+                .map(ExecutableElement.class::cast)
+                .filter(element -> element.getThrownTypes().stream().allMatch(this::isRuntimeException)
+                        && !element.getParameters().isEmpty())
+                .map(element -> {
+                    List<CodeBlock> args = new ArrayList<>();
+                    for (VariableElement parameter : element.getParameters()) {
+                        TypeName parameterTypeName = TypeName.get(parameter.asType());
+                        if (parameterTypeName.equals(ClassName.get(UndertowRuntime.class))) {
+                            args.add(CodeBlock.of("runtime"));
+                        } else if (parameterTypeName.equals(ClassName.get(Endpoint.class))) {
+                            args.add(CodeBlock.of("this"));
+                        } else {
+                            return null;
+                        }
+                    }
+                    return CodeBlock.of("new $T($L)", TypeName.get(declaredType), CodeBlock.join(args, ", "));
+                })
```

Comment:
We could require all constructors to have both `UndertowRuntime` and `Endpoint` parameters in a particular order, but:
- This would require users to write constructors with unused parameters, which is a bit strange.
- If we ever support additional parameters in the future, we'd have to do something like this anyways.

Reply:
👍 it keeps things simple for folks implementing these